### DISCUSSION TITLE
Issues in OpenLitterMap web application

### DISCUSSION THREAD

**Discussion by farisyasabri** [OpenLitterMap/openlittermap-web#629] [General]
Recently my team and I have conducted several testing on your application and we found several errors. 

**Issue in Create Cleanup Features**

In Create Cleanup feature, we have conducted use case testing. However, there is multiple issues in this features:
1. System allows the user to create a cleanup with the number of characters.
![image](https://github.com/OpenLitterMap/openlittermap-web/assets/74453148/a983f60b-254f-4a59-89dd-043d9108938e)

2.System allows the user to create a cleanup with a date that has passed.
![image](https://github.com/OpenLitterMap/openlittermap-web/assets/74453148/af4872c8-53c5-414b-afe1-9cb8411a4de8)

3. System has incorrect validation for the date field in the create cleanup form. The error message “The date field is required” should be hidden once a valid date is input but the system still shows the error message when the date field is correctly inputted.
![image](https://github.com/OpenLitterMap/openlittermap-web/assets/74453148/dbf92912-d377-404c-94d1-629f6d288723)

4. System allows the user to submit the form for Date: “abc”
![image](https://github.com/OpenLitterMap/openlittermap-web/assets/74453148/5bcb7721-26b2-4240-8924-ed3c02406663)

**Issue in Join Cleanup Feature**
1. System does not show any message success message and adds the user to the cleanup.

**Issues in View Leaderboard Feature**
1. The system does not display any 100 ranked users
![WhatsApp Image 2023-06-17 at 21 27 15](https://github.com/OpenLitterMap/openlittermap-web/assets/74453148/4b2a5080-ab5b-4bb3-8b1d-0340563e2367)

**Issues in Filter Litter Types Feature**
1. The system does not display any users filtered litter in a pie chart
![image](https://github.com/OpenLitterMap/openlittermap-web/assets/74453148/1648b76e-d1c4-4ba4-a2ff-7e3cc15f4d3b)

**Issues in View Profile feature**
1. The system does not display any data of verified photos depicted in a line graph in the next section
![image](https://github.com/OpenLitterMap/openlittermap-web/assets/74453148/3d1e32b4-b518-4b19-a9bb-b5c5756e01ef)

**Issues in Create Team feature**
1. System creates a team and redirects user to the team dashboard and not display a warning message “The team identifier needs to have at least a number and uppercase character”.
![image](https://github.com/OpenLitterMap/openlittermap-web/assets/74453148/ea95004a-d3d5-4df0-b883-c52d49c87d50)
![image](https://github.com/OpenLitterMap/openlittermap-web/assets/74453148/e317cb32-52d5-4e1c-aeea-20daabd0fb58)

**Issues in Forgot Password Feature**
1. System displays a success message and inform the user that the password reset link will be sent if the email exists and not display an error message “The email does not exist in our record”.
![image](https://github.com/OpenLitterMap/openlittermap-web/assets/74453148/bb777b87-8ceb-4248-880e-8a224adfb791)

**Comment by xlcrr** [comment]
Hey @farisyasabri thanks for sharing these insights

I am aware that OpenLitterMap has many bugs, you can find out about some of the things we are working on here https://trello.com/b/RkGmQyEh/openlittermap-web

If you would like to help us fix them we would love some help!

Stay tuned for future releases

Thanks,

Seán
